modules/markup/html.go

```diff
@@ -907,6 +911,23 @@ func issueIndexPatternProcessor(ctx *RenderContext, node *html.Node) {
 	}
 }
 
+func commitCrossReferencePatternProcessor(ctx *RenderContext, node *html.Node) {
+	next := node.NextSibling
+
+	for node != nil && node != next {
+		found, ref := references.FindRenderizableCommitCrossReference(node.Data)
+		if !found {
+			return
+		}
+
+		reftext := ref.Owner + "/" + ref.Name + "@" + base.ShortSha(ref.CommitSha)
+		link := createLink(util.URLJoin(setting.AppURL, ref.Owner, ref.Name, "commit", ref.CommitSha), reftext, "commit")
```

Comment:
Absolute link? Aren't we trying at the moment to replace them with relative links?

Reply:
Use `setting.AppSubURL` instead.